pkg/oc/cli/cmd/set/deployment_strategy.go

```diff
@@ -0,0 +1,539 @@
+	cmd.Flags().StringArrayVarP(&options.Environment, "environment", "e", options.Environment, "Environment variables to use in the custom deployment pod")
+
+	cmd.Flags().BoolVar(&options.Recreate, "recreate", options.Recreate, "Set a Recreate strategy")
+	cmd.Flags().BoolVar(&options.Rolling, "rolling", options.Rolling, "Set a Rolling strategy")
+	cmd.Flags().BoolVar(&options.Custom, "custom", options.Custom, "Set a Custom strategy")
+	cmd.Flags().BoolVar(&options.OnDelete, "on-delete", options.OnDelete, "Set a OnDelete strategy")
+
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set strategy will NOT contact api-server but run locally.")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter resources")
+	cmd.Flags().BoolVar(&options.All, "all", options.All, "If true, select all resources in the namespace of the specified resource types")
+
+	cmd.MarkFlagFilename("filename", "yaml", "yml", "json")
+	kcmdutil.AddDryRunFlag(cmd)
+
+	return cmd
+}
+
+func (o *DeploymentStrategyOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string) error {
+	resources := args
+	if i := cmd.ArgsLenAtDash(); i != -1 {
+		resources = args[:i]
+		o.Command = args[i:]
+	}
+	if len(o.Filenames) == 0 && len(args) < 1 {
+		return kcmdutil.UsageErrorf(cmd, "one or more resources must be specified as <name> or dc/<name>")
+	}
+
+	cmdNamespace, explicit, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.Cmd = cmd
+
+	mapper, _ := f.Object()
+	o.Builder = f.NewBuilder(!o.Local).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(explicit, &resource.FilenameOptions{Recursive: false, Filenames: o.Filenames}).
+		Flatten()
+
+	if !o.Local {
+		o.Builder = o.Builder.
+			SelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, resources...)
+	}
+
+	o.Output = kcmdutil.GetFlagString(cmd, "output")
+	o.PrintObject = func(infos []*resource.Info) error {
+		return f.PrintResourceInfos(cmd, o.Local, infos, o.Out)
+	}
+
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = kcmdutil.GetFlagString(cmd, "output") == "name"
+	o.Mapper = mapper
+
+	return nil
+}
+
+// Validate validates the options for different strategies.
+func (o *DeploymentStrategyOptions) Validate() error {
+	switch {
+	case o.Recreate:
+		return o.validateRecreate()
+	case o.Rolling:
+		return o.validateRolling()
+	case o.OnDelete:
+		return o.validateOnDelete()
+	case o.Custom:
+		return o.validateCustom()
+	default:
+		return kcmdutil.UsageErrorf(o.Cmd, "strategy must be selected")
+	}
+}
+
+// TODO: Add validation of params here? It might be overlapping with API validation.
+func (o *DeploymentStrategyOptions) validateRecreate() error {
+	if o.Cmd.Flags().Changed("max-surge") {
```

Comment:
it is tricky because we don't have the list of resources at this point. i guess this validation is just minimal pre-flight check, you will get API validation errors anyway. maybe we should nuke this validation

Reply:
you would silently miss flags that are set on CLI and ignored in that particular strategy+resource